pkg/deploy/registry/instantiate/strategy.go

```diff
@@ -0,0 +1,73 @@
+package instantiate
+
+import (
+	"reflect"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+)
+
+type strategy struct {
+	runtime.ObjectTyper
+}
+
+var Strategy = strategy{kapi.Scheme}
+
+func (strategy) NamespaceScoped() bool {
+	return true
+}
+
+func (strategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (strategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (strategy) GenerateName(base string) string {
+	return base
+}
+
+// PrepareForCreate is a no-op for the instantiate endpoint.
+func (strategy) PrepareForCreate(obj runtime.Object) {
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by the instantiate endpoint.
+func (strategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDc := obj.(*api.DeploymentConfig)
```

Comment:
I don't need to call explicitly these handlers, do I? The endpoint works fine and it lets me update the spec and the part of the status that I want to update so I guess this works...